src/rust/lib/graph-editor/src/component/node/port/output.rs

```diff
@@ -0,0 +1,682 @@
+        fn set_grow(&self, grow_value:f32) {
+            self.grow.set(grow_value)
+        }
+
+        fn set_opacity(&self, opacity:f32) {
+            self.opacity.set(opacity)
+        }
+    }
+}
+
+/// Implements a simplified version of the multi_port_area::Shape shape for the case where there is
+/// only a single output port.
+pub mod single_port_area {
+    use super::*;
+    use ensogl::display::shape::*;
+
+    ensogl::define_shape_system! {
+        (style:Style, grow:f32, opacity:f32) {
+            let overall_width  : Var<Distance<Pixels>> = "input_size.x".into();
+            let overall_height : Var<Distance<Pixels>> = "input_size.y".into();
+
+            let base_shape_data = port_base_shape_data(&overall_width, &overall_height, &grow);
+            let BaseShapeData{ port_area,hover_area, .. } = base_shape_data;
+
+            // FIXME: Use colour from style and apply transparency there.
+            let color     = Var::<color::Rgba>::from("srgba(0.25,0.58,0.91,input_opacity)");
+            let port_area = port_area.fill(color);
+
+            (port_area + hover_area).into()
+        }
+    }
+
+    impl PortShapeApi for Shape {
+        fn set_grow(&self, grow_value:f32) {
+            self.grow.set(grow_value)
+        }
+
+        fn set_opacity(&self, opacity:f32) {
+            self.opacity.set(opacity)
+        }
+    }
+}
+
+/// Helper enum that handles the distinction between a single shape output area and a multi port
+/// output area.
+#[derive(Clone,Debug)]
+enum ShapeView {
+    Single { view  : component::ShapeView<multi_port_area::Shape>      },
+    Multi  { views : Vec<component::ShapeView<multi_port_area::Shape>> },
+}
+
+impl ShapeView {
+    /// Constructor.
+    fn new(number_of_ports:NonZeroU32, logger:&Logger, scene:&Scene) -> Self {
+        let number_of_ports = number_of_ports.get();
+        if number_of_ports == 1 {
+            ShapeView::Single { view: component::ShapeView::new(&logger,&scene) }
+        } else {
+            let mut views = Vec::default();
+            let number_of_ports = number_of_ports as usize;
+            views.resize_with(number_of_ports,|| component::ShapeView::new(&logger,&scene));
+            ShapeView::Multi { views }
+        }
+    }
+
+    /// Set up the frp for all ports.
+    fn init_frp(&self, port_frp:PortFrp) {
+        match self {
+            ShapeView::Single {view}  => init_port_frp(&view, PortId::new(0),port_frp),
+            ShapeView::Multi  {views} => {
+                views.iter().enumerate().for_each(|(index,view)| {
+                    init_port_frp(&view, PortId::new(index),port_frp.clone_ref())
+                })
+            }
+        }
+    }
+
+    /// Resize all the port output shapes to fit the new layout requirements for thr given
+    /// parameters.
+    fn update_shape_layout_based_on_size_and_gap(&self, size:Vector2<f32>, gap_width:f32) {
+        match self {
+            ShapeView::Single { view }   => {
```

Comment:
alignment of `=>` - just do not align it to anything, as the line above or below do not contain `=>`